### DISCUSSION TITLE
Datepicker

### DISCUSSION THREAD

**Discussion by orenzav** [israel-dryer/ttkbootstrap#554] [Q&A]
Hi, I'm trying to make an interface that's keyboard only driven I have the key bindings working for most widgets that I need but I can't get them working for the datepicker. I'm only using the arrow keys, enter and tab on the keyboard so need the calendar popup to launch and allow me to navigate to a date on it using tab/arrows/enter for select. It doesn't seem like keyboard bindings are supported though, does anyone know any different? I can tab into the datepickerwidget but not launch the popup. If I do launch it (using a mouse, not an option going forward) then I can tab to the year change arrows but not the date section to select a date.

**Comment by TheDemonGuard** [comment]
> Hi, I'm trying to make an interface that's keyboard only driven I have the key bindings working for most widgets that I need but I can't get them working for the datepicker. I'm only using the arrow keys, enter and tab on the keyboard so need the calendar popup to launch and allow me to navigate to a date on it using tab/arrows/enter for select. It doesn't seem like keyboard bindings are supported though, does anyone know any different? I can tab into the datepickerwidget but not launch the popup. If I do launch it (using a mouse, not an option going forward) then I can tab to the year change arrows but not the date section to select a date.

Please provide working minimum reproducible demo/sample script.
And any necessary steps/instructions to reproduce the issue.
This will be helpful for anyone who will be interested in investigating the issue.

**Comment by israel-dryer** [comment]
@orenzav , I'll look into it. Thanks for bring it up.

**Comment by israel-dryer** [comment]
@orenzav I've updated the calendar styles to enable and display keyboard focus traversal.  Should be in the next release. 

<img width="250" height="275" alt="Image" src="https://github.com/user-attachments/assets/af9b23ea-b6fa-4526-9a94-d6289d668007" />

**Reply by orenzav** [reply]
Thank you :)
